src/ips.rs

```diff
@@ -63,87 +72,60 @@ pub const ERR_PARSE_IP: &str = "failed to parse IP address";
 const ERR_GET_DEGREE: &str = "failed to get degree";
 const ERR_GET_EIGENVECTOR: &str = "failed to get eigenvector";
 
-#[derive(Default, Clone)]
+#[derive(Default, Clone, Copy)]
 struct NormalizationFactors {
     min: f64,
     max: f64,
 }
 
 impl Ips {
     pub fn new(config: IPSConfiguration) -> Ips {
-        Ips {
-            config,
-            ..Default::default()
-        }
+        Ips { config }
     }
 
     /// Generate peer list - main function with The Algorithm
     /// It needs state and agraph to be passed as parameters which need to be correlated with
     /// the crawler's state and agraph (and with each other), so the indexes saved in the
```

Comment:
I think we should phase out use of the word `agraph`, since it's now gone from `spectre`.  I would suggest using `indices` or `indices struct` instead, which refers to `Vec<Vec<usize>>`.   
Also, in the future, maybe we should make `Indices` or `NodeIndices` an explicit type for this structure, defined in `ziggurat-core/ziggurat-core-crawler`?  [outside the scope of this PR.]

Reply:
Removed agraph from comment. 

It could be a good way to move `Indices` to the zg-core just to ensure all apps are using the same structures.

Reply:
> It could be a good way to move Indices to the zg-core just to ensure all apps are using the same structures.

we could move such struct it to spectre, since those term originate from that repo, e.g.:
```rust
    pub fn get_adjacency_indices(&mut self) -> Vec<Vec<usize>> {
```

`usize` ===> `Index`
`Vec<Vec<usize>> `  -> `Vec<Vec<Index>>` ===> `Indices` (maybe)

then we can reference that struct definition from our repos

Reply:
As agreed on Slack, we'd move it to ziggurat-core.